src/components/ButtonClientMobile.jsx

```diff
@@ -0,0 +1,35 @@
+/* eslint-env browser */
+
+import styles from '../styles/buttonclient'
+
+import { translate } from '../lib/I18n'
+
+import React, { Component } from 'react'
+
+import classNames from 'classnames'
+
+class ButtonClientMobile extends Component {
+  constructor (props) {
+    super(props)
+    const seenAd = localStorage.getItem('app_ad')
+    this.state = {
+      seen: seenAd ? JSON.parse(seenAd) : false
+    }
```

Comment:
To clarify the algorithm, you shall write this way:

```js
this.state = {
    seen: JSON.parse(localStorage.getItem('app_ad')) || false
}
```

Then `this.state.seen` takes the value of `JSON.parse(localStorage.getItem('app_ad'))` or `false` if it does not exist.

Another way to write this is with a function:

```js
this.state = {
    seen: getSeenAdState()
}
```

with `getSeenAdState()` as:

```js
function getSeenAdState () {
    const seenAd = localStorage.getItem('app_ad')
    return seenAd ? JSON.parse(seenAd) : false 
}
```

This way the function is responsible to `getSeenAdState` wherever and however the value is stored, retrieved and returned. SRP.